src/v2/coins/abstractUtxoCoin.ts

```diff
@@ -833,46 +826,56 @@ class AbstractUtxoCoin extends BaseCoin {
-        const prevOutScript = bitcoin.script.witnessScriptHash.output.encode(witnessScriptHash);
-        // we are faking a signature script for the verification
-        signatureScript = bitcoin.script.compile([prevOutScript]);
+      if (isNativeSegwitInput) {
+        inputClassification = bitcoin.script.classifyWitness(bitcoin.script.compile(decompiledSigScript), true);
+      } else {
+        inputClassification = bitcoin.script.classifyInput(currentInput.script, true);
       }
-    }
+    } else {
+      inputClassification = bitcoin.script.classifyInput(currentInput.script, true);
+      decompiledSigScript = bitcoin.script.decompile(currentInput.script);
 
-    const inputClassification = bitcoin.script.classifyInput(signatureScript, true);
-    if (inputClassification !== bitcoin.script.types.P2SH) {
-      if (inputClassification === bitcoin.script.types.P2PKH) {
-        const [signature, publicKey] = decompiledSigScript;
-        const publicKeys = [publicKey];
-        const signatures = [signature];
+    }
 
-        const pubScript = bitcoin.script.pubKeyHash.output.encode(bitcoin.crypto.hash160(publicKey));
-        return { signatures, publicKeys, isSegwitInput, inputClassification, pubScript: pubScript };
-      }
+    if (inputClassification === bitcoin.script.types.P2PKH) {
+      const [signature, publicKey] = decompiledSigScript;
+      const publicKeys = [publicKey];
+      const signatures = [signature];
+      const pubScript = bitcoin.script.pubKeyHash.output.encode(bitcoin.crypto.hash160(publicKey));
+
+      return { signatures, publicKeys, isSegwitInput, inputClassification, pubScript };
+    } else if (inputClassification === bitcoin.script.types.P2SH
+        || inputClassification === bitcoin.script.types.P2WSH) {
+      // Note the assumption here that if we have a p2sh or p2wsh input it will be multisig (appropriate because the
+      // BitGo platform only supports multisig within these types of inputs). Signatures are all but the last entry in
+      // the decompiledSigScript. The redeemScript/witnessScript (depending on which type of input this is) is the last
+      // entry in the decompiledSigScript (denoted here as the pubScript). The public keys are the second through
+      // antepenultimate entries in the decompiledPubScript. See below for a visual representation of the typical 2-of-3
+      // multisig setup:
+      //
+      // decompiledSigScript = 0 <sig1> <sig2> <pubScript>
+      // decompiledPubScript = 2 <pub1> <pub2> <pub3> 3 OP_CHECKMULTISIG
+      const signatures = decompiledSigScript.slice(0, -1);
+      const pubScript = _.last(decompiledSigScript);
+      const decompiledPubScript = bitcoin.script.decompile(pubScript);
+      // the second through antepenultimate entries
+      const publicKeys = decompiledPubScript.slice(1, -2);
+
+      return { signatures, publicKeys, isSegwitInput, inputClassification, pubScript };
```

Comment:
please order it so the common fields `isSegwitInput` and `inputClassification` come first

also in the previous branch

Reply:
Done